File: Other Problems/fibonacciNumber.py
def fibonacciNumber(n, a, b):
    c = 0
    for i in range(2, n+1):
        c = a + b
        a = b
        b = c
    return c


n = 10
a = 0
b = 1

print("Fibonacci Sequence -> ", end=" ")
for i in range(n):
    print(fibonacciNumber(i, a, b), end=" ")

print("\nNth Fibonacci -> ", fibonacciNumber(n-1, a, b))

File: Other Problems/fibonacciNumber.cpp
#include <iostream>
using namespace std;

int fibonacciNumber(int n, int a, int b)
{
    int c = 0;
    for (int i = 2; i <= n; ++i)
    {
        c = a + b;
        a = b;
        b = c;
    }
    return c;
}

int main()
{
    int n = 10;
    int a = 0, b = 1;

    cout << "Fibonacci Sequence -> ";
    for (int i = 0; i < n; ++i)
    {
        cout << fibonacciNumber(i, a, b) << " ";
    }
    cout << endl;

    cout << "Recursive -> " << fibonacciNumber(n - 1, a, b) << endl;

    return 0;
}
